core/src/main/java/tachyon/web/WebInterfaceWorkerBlockInfoServlet.java

```diff
@@ -0,0 +1,161 @@
+
+import tachyon.StorageDirId;
+import tachyon.StorageLevelAlias;
+import tachyon.TachyonURI;
+import tachyon.client.TachyonFS;
+import tachyon.conf.TachyonConf;
+import tachyon.master.BlockInfo;
+import tachyon.thrift.ClientFileInfo;
+import tachyon.thrift.FileDoesNotExistException;
+import tachyon.worker.WorkerStorage;
+import tachyon.worker.hierarchy.StorageDir;
+
+/**
+ * Servlet that provides data for displaying block info of a worker.
+ */
+public class WebInterfaceWorkerBlockInfoServlet extends HttpServlet {
+  private static final long serialVersionUID = 4148506607369321012L;
+  private final transient WorkerStorage mWorkerStorage;
+  private final TachyonConf mTachyonConf;
+
+  public WebInterfaceWorkerBlockInfoServlet(WorkerStorage workerStorage, TachyonConf conf) {
+    mWorkerStorage = workerStorage;
+    mTachyonConf = conf;
+  }
+
+  /**
+   * Populates attributes before redirecting to a jsp.
+   *
+   * @param request The HttpServletRequest object
+   * @param response The HttpServletReponse object
+   * @throws ServletException
+   * @throws IOException
+   */
+  @Override
+  protected void doGet(HttpServletRequest request, HttpServletResponse response)
+      throws ServletException, IOException {
+    request.setAttribute("fatalError", "");
+
+    TachyonFS tachyonClient = TachyonFS.get(mTachyonConf);
+    Map<Integer, UiFileInfo> uiFileInfoMap = getUiFileInfoMap(tachyonClient);
+    List<UiFileInfo> uiFileInfos = new ArrayList<UiFileInfo>(uiFileInfoMap.values());
+    Collections.sort(uiFileInfos, UiFileInfo.PATH_STRING_COMPARE);
+
+    String filePath = request.getParameter("path");
+    if (!(filePath == null || filePath.isEmpty())) {
+      // Display file block info
+      try {
+        ClientFileInfo fileInfo = tachyonClient.getFileStatus(-1, new TachyonURI(filePath));
+        if (fileInfo == null || !uiFileInfoMap.containsKey(fileInfo.getId())) {
+          throw new FileDoesNotExistException(filePath);
+        }
+        UiFileInfo uiFileInfo = uiFileInfoMap.get(fileInfo.getId());
+        request.setAttribute("fileBlocksOnTier", uiFileInfo.getBlocksOnTier());
+        request.setAttribute("blockSizeByte", uiFileInfo.getBlockSizeBytes());
+        request.setAttribute("path", filePath);
+
+        getServletContext().getRequestDispatcher("/worker/viewFileBlocks.jsp").forward(request,
+            response);
+        return;
+      } catch (FileDoesNotExistException fdne) {
+        request.setAttribute("fatalError", "Error: Invalid Path " + fdne.getMessage());
+        getServletContext().getRequestDispatcher("/worker/blockInfo.jsp").forward(request,
+            response);
+        return;
+      } catch (IOException ie) {
+        request.setAttribute("invalidPathError", "Error: File " + filePath + " is not available "
+            + ie.getMessage());
+        getServletContext().getRequestDispatcher("/worker/blockInfo.jsp").forward(request,
+            response);
+        return;
+      }
+    }
```

Comment:
In the case of `filePath` set, can we avoid populating the `uiFileInfoMap` since we are getting the fileInfo anyway? That seems like an expensive operation.